miniscope_io/cli/stream.py

```diff
@@ -93,3 +115,29 @@ def test(ctx: click.Context, source: Path, profile: bool, **kwargs: dict) -> Non
     os.environ["PYTEST_OKDEV_DATA_FILE"] = str(source)
 
     ctx.forward(capture)
+
+
+def get_unique_stempath(base_output: Path) -> Path:
```

Comment:
The `while _any_stem_exists(candidate_stem):` loop below does the numbering thing (though not formatted to 5 digits), but are you thinking of something in addition?

Having this in `.env` makes sense too, but I couldn't fully understand the following. Do you mean making another model that defines the string formats used for file/folder exports?
>the model would declare the kinds of things that would be available in the template string, and we could either use python string interpolation or jinja for this. I'm thinking this would be a good mechanism in general, since it's a super common thing to want to do, and also i want to introduce the concept so we can use it in the metadata schema too, where people can declare a template for their experimental metadata as part of their config.

Reply:
We might want to discuss user interfaces in general, including how this I/O will be used (if this is just a module or a stand-alone thing, etc. You guys might have chatted about this in the last lab meeting). Also, I'm randomly adding stuff without much knowledge about later stages (the only thing I've used before is CaImAn), and formats easy to connect to these should probably be considered too.